test/IDE/complete_swift_key_path_inference.swift

```diff
@@ -0,0 +1,41 @@
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=TYPE_INFER_DOT | %FileCheck %s -check-prefix=PERSONTYPE-INFER-DOT
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=TYPE_INFER_DOT_OPTIONAL | %FileCheck %s -check-prefix=PERSONTYPE-INFER-DOT-OPT
+// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=TYPE_INFER_DOT_OPTIONAL_UNWRAP | %FileCheck %s -check-prefix=PERSONTYPE-INFER-DOT-OPT-UNWRAP
+
+class Person {
+    var name: String
+    var friends: [Person] = []
+    var bestFriend: Person? = nil
+    var itself: Person { return self }
+    init(name: String) {
+        self.name = name
+    }
+    func getName() -> String { return name }
+    subscript(_ index: Int) -> Int { get { return 1} }
+}
+
+extension Optional {
+    var optMember: String { "member" }
+}
+
+let _ : KeyPath<Person, String> = \.#^TYPE_INFER_DOT^#
+// PERSONTYPE-INFER-DOT: Begin completions, 5 items
+// PERSONTYPE-INFER-DOT-NEXT: Decl[InstanceVar]/CurrNominal:      name[#String#]; name=name
+// PERSONTYPE-INFER-DOT-NEXT: Decl[InstanceVar]/CurrNominal:      friends[#[Person]#]; name=friends
+// PERSONTYPE-INFER-DOT-NEXT: Decl[InstanceVar]/CurrNominal:      bestFriend[#Person?#]; name=bestFriend
+// PERSONTYPE-INFER-DOT-NEXT: Decl[InstanceVar]/CurrNominal:      itself[#Person#]; name=itself
+// PERSONTYPE-INFER-DOT-NEXT: Decl[Subscript]/CurrNominal:        [{#(index): Int#}][#Int#]; name=[index: Int]
+
+let _ : KeyPath<Person?, String> = \.#^TYPE_INFER_DOT_OPTIONAL^#
+// PERSONTYPE-INFER-DOT-OPT: Begin completions, 4 items
+// PERSONTYPE-INFER-DOT-OPT-NEXT: Decl[InstanceVar]/CurrNominal/IsSystem: unsafelyUnwrapped[#Person#]; name=unsafelyUnwrapped
```

Comment:
Would it be possible to filter out members whose type does not match KeyPath's value type? For example, `unsafelyUnwrapped` returns `Person` whereas the KeyPath's value type is `String` so it's not a valid suggestion.